### PR TITLE
Fix script file issues and move script classes into namespace

### PR DESCRIPTION
**Pull request by mortenhauberg**
`Elastica\Script` and `Elastica\ScriptFile` extends `Elastica\AbstractScript`.
There are `instanceof` checks and type-hints for `Elastica\Script` in different classes.
I've changed the checks and type-hints to `Elastica\AbstraceScript`, and made `Elastica\ScriptFile` extend `Elastica\Script`.

@ruflin suggested to move the scripting classes into a namespace of its own. 
We decided to keep the old classes, and have them extend the new classes in `Elastica\Script` for BC reasons.

Old|New
------------ | -------------
`Elastica\AbstractScript` |`Elastica\Script\AbstractScript`
`Elastica\Script` |`Elastica\Script\Script`
`Elastica\ScriptFile` |`Elastica\Script\ScriptFile`
`Elastica\ScriptFields` |`Elastica\Script\ScriptFields`

Please let me know if anything needs changes :)

### CODE REVIEW

**Comment by ruflin** [comment]
@mortenhauberg Can you update the CHANGELOG.md?

File: lib/Elastica/AbstractScript.php [hunk 1]
```diff
@@ -1,27 +1,10 @@
...
 namespace Elastica;
 
+use Elastica\Script\AbstractScript as BaseAbstractScript;
 /**
- * Base class for Script object.
- *
- * @author Nicolas Assing <nicolas.assing@gmail.com>
- *
- * @link https://www.elastic.co/guide/en/elasticsearch/reference/current/modules-scripting.html
+ * Kept for BC reasons.
```
Comment (ruflin) [inline lib/Elastica/AbstractScript.php]: Can we add a @deprecated message here? Check some other files which are deprecated for some examples.

**Comment by ruflin** [comment]
LGTM. It would be nice to have at least a deprecated message in the docs in all classes which are deprecated.  Here is an example: https://github.com/ruflin/Elastica/blob/f124e598228a6b8a6c249fde6afc07ea7eb1a771/lib/Elastica/Query/Bool.php

**Comment by mortenhauberg** [comment]
:+1: 
Something like this?

**Comment by ruflin** [comment]
@mortenhauberg Merged. Thanks. Perhaps in the future we can also trigger an error so it shows up in the logs: https://github.com/ruflin/Elastica/blob/f124e598228a6b8a6c249fde6afc07ea7eb1a771/lib/Elastica/Query/Bool.php#L4

**Comment by mortenhauberg** [comment]
Thank you. I'll create a PR for triggering errors next week :)

**Comment by ruflin** [comment]
:+1:

**Pull request merged**